docs/source/aggregation.rst

```diff
@@ -0,0 +1,176 @@
+
+    @Project.operation
+    def op2(job):
+        pass
+
+    if __name__ == '__main__':
+        Project().main()
+
+If :class:`~flow.aggregator` is used with the default arguments, an aggregate of all the jobs present in the project will be created.
+In the above example, ``op1`` can be referred to as an *aggregate operation* where all the jobs present in the project are passed as arbitraty arguments (or ``*args``) and ``op2`` is a *normal operation* where only a single job is passed as an argument.
+
+.. note::
+
+    For an aggregate operation, all the other functionalities like :class:`~flow.FlowProject.pre`, :class:`~flow.FlowProject.post`, callable directives, etc., are required to take the same number of jobs as the operation as arguments.
+
+.. _types_of_aggregation:
+
+Types of Aggregation
+====================
+
+Currently, **signac-flow** allows users to aggregate jobs by:
+
+- Grouping them on state point key, an iterable of state point keys whose values define the groupings, or an arbitrary callable of :class:`~signac.contrib.job.Job`.
+- Generating aggregates of a given size.
+- Using custom aggregator function when greater flexibility is needed.
+
+Group By
+---------
+
+:class:`~flow.aggregator.groupby` allows users to aggregate jobs by grouping them on state point key, an iterable of state point keys whose values define the groupings, or an arbitrary callable of :class:`~signac.contrib.job.Job`.
+
+.. code-block:: python
+
+    @aggregator.groupby('temperature')
+    @Project.operation
+    def op3(*jobs):
+        pass
+
+In the above example, the jobs will get aggregated based on the state point **temperature**.
+So, all the jobs having the same value of **temperature** in their state point will be aggregated together.
+
+Groups Of
+---------
+
+:class:`~flow.aggregator.groupsof` allows users to aggregate jobs by generating aggregates of a given size.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2)
+    @Project.operation
+    def op4(job1, job2=None):
+        pass
+
+In the above example, the jobs will get aggregated in groups of 2 and hence, up to two jobs will be passed as arguments at once.
+
+.. note::
+
+    In case the number of jobs in the project is odd, there will be one aggregate containing only a single job and hence users should be careful while passing non-default arguments in an *aggregate operation*.
+
+Sorting jobs for aggregation
+----------------------------
+
+**signac-flow** allows users to define the sorting order of jobs before creating the aggregates with the help of ``sort_by`` parameter and the sorting order can be defined with the help of ``sort_ascending`` parameter.
+By default, when no ``sort_by`` parameter is specified, the order of the jobs will be decided by the order in which the jobs are iterated in a **signac** project.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2, sort_by='temperature', sort_ascending=False)
+    @Project.operation
+    def op5(job1, job2):
```

Comment:
Do we need to use `*jobs` or `job2=None` to support a final aggregate with one job?
(I also worry that showing examples with `job1, job2` will be more confusing than `*jobs`.)

Reply:
Since we've made our point of using non default arguments carefully, I think we should go with `*jobs` here.